tools/autograd/templates/VariableType.cpp

```diff
@@ -289,39 +289,65 @@ static void check_inplace(const Tensor& tensor) {
   }
 }
 
-static void rebase_history(Tensor& tensor, std::shared_ptr<Function> grad_fn, int output_nr=0) {
-  if (!tensor.defined()) {
-    return;
+static void check_output_args(const char* name, TensorList tensors) {
+  // Our logic for modifications to views only works for in-place functions,
+  // not out=... functions. Checks that no output arguments are views.
+  for (auto& tensor : tensors) {
+    if (tensor.defined()) {
+      auto& var = static_cast<const Variable&>(tensor);
+      if (var.is_view()) {
+        at::runtime_error(
+          "%s(): output arguments (out=...) must not be views in "
+          "differentiable functions", name);
+      }
+    }
   }
-  auto& var = static_cast<Variable&>(tensor);
-  if (grad_fn) {
+}
+
+static void rebase_history(Tensor& tensor, std::shared_ptr<Function> grad_fn) {
+  if (grad_fn && tensor.defined()) {
+    auto& var = static_cast<Variable&>(tensor);
     grad_fn->num_inputs = 1;
-    var.rebase_history(output_nr, std::move(grad_fn));
+    var.rebase_history(0, std::move(grad_fn));
+  }
+}
+
+static void rebase_history(TensorList tensors, std::shared_ptr<Function> grad_fn) {
```

Comment:
Yeah, all the functions with multiple outputs call it when called with the `out=...` syntax. We don't support rebasing views with non-zero output_nr, which is why we emit calls to `check_output_args` to ensure that none of the output arguments are views.

Eventually, it would be nice to support that, but that's low-priority right now.